sklearn/kernel_approximation.py

```diff
@@ -658,7 +658,10 @@ def __init__(self, *, sample_steps=2, sample_interval=None):
         self.sample_interval = sample_interval
 
     def fit(self, X, y=None):
-        """Set the parameters.
+        """Do nothing and return the estimator unchanged.
+
+        This method is just there to implement the usual API and hence
```

Comment:
Actually, this is not no-op anymore because we do parameter validation.
I think we need to change the documentation in this regard.

We should also add some documentation to encourage using `fit_transform`
instead of only `transform` to benefit from parameter validation. I assume that this could lend in the "Notes" section.